### DISCUSSION TITLE
networkx place nodes with specific position

### DISCUSSION THREAD

**Discussion by samahwaleed** [networkx/networkx#5230] [Q&A] [answered]
I want to add multiple types of nodes in different position, I used the following code:

```
g = torch_geometric.utils.to_networkx(data, to_undirected=True)
plt.figure(len(data))
nx.draw(g, pos=posDict, with_labels=True)
plt.show()
```

where the `data `looks like this:
`Data(pos=[3, 2], x=[1080, 1920, 3], c=[3], edge_index=[2, 6], edge_attr=[6, 1])
`

And the posDict has the nodes and the distance between them:

```
{0: array([0.740365, 0.126852]), 1: array([0.755729, 0.473148]), 2: array([0.917969, 0.469907])}
{0: array([0.107552, 0.806944]), 1: array([0.29375 , 0.373148]), 2: array([0.576042, 0.358333]), 3: array([0.434115, 0.376852]), 4: array([0.0471354, 0.390741 ]), 5: array([0.161458, 0.376389]), 6: array([0.697135, 0.318056])}
.....
```

By doing:
`print(g.nodes())`

I get:

> [0, 1, 2, 3, 4, 5, 6, 7] [0, 1, 2, 3, 4, 5] [0, 1, 2, 3, 4, 5] [0, 1,2, 3, 4, 5, 6] [0, 1, 2, 3, 4, 5, 6] [0, 1, 2] [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] [0, 1, 2, 3, 4, 5] [0, 1, 2] [0, 1, 2, 3, 4, 5] [0, 1, 2, 3, 4, 5, 6] [0, 1, 2, 3, 4, 5, 6] [0, 1, 2, 3, 4, 5, 6, 7] [0, 1, 2, 3] .....

I got this error:

> NetworkXError: Node 3 has no position.

**Comment by MridulS** [comment, marked as answer]
The [pos](https://networkx.org/documentation/stable/reference/generated/networkx.drawing.nx_pylab.draw.html#networkx.drawing.nx_pylab.draw) dictionary argument in `nx.draw` needs to have positions keyed by node.

Does `posDict[3]` return anything ?

**Reply by samahwaleed** [reply]
Thank you Sir. It's working now
